### PR TITLE
php@*.*: fix deprecation and disable dates

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This brings it in line with the information that's available at: https://www.php.net/supported-versions.php

### CODE REVIEW

File: Formula/php@8.0.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ class PhpAT80 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2023-11-26", because: :versioned_formula
+  deprecate! date: "2022-11-26", because: :versioned_formula
```
Comment (Bo98) [inline Formula/php@8.0.rb:26]: The page you linked says security support until 2023?

Comment (SMillerDev) [reply Formula/php@8.0.rb:26]: Yes, so it's deprecated from 2022 and EOL in 2023.

Comment (Bo98) [reply Formula/php@8.0.rb:26]: Ok, that makes sense. I was comparing dates with the other formulae so it confused me a little.

**Review by Bo98 — APPROVED** [review]

File: Formula/php@8.0.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/php@8.0.rb:26]: How do you feel about adding a commented disable for that? Ideally we can define both in one go, but unfortunately it doesn't work that way.

Comment (Bo98) [reply Formula/php@8.0.rb:26]: A commented disable makes sense to me.

**Pull request closed without merging**